meta2v2/meta2_utils.c

```diff
@@ -1999,6 +2000,235 @@ m2_generate_beans(struct oio_url_s *url, gint64 size, gint64 chunk_size,
 	}
 }
 
+enum _content_broken
+{
+	NONE,
+	REPARABLE,
+	IRREPARABLE
+};
+
+struct _check_content
+{
+	gint64 size;
+	gint last_pos;
+	GString *present_pos;
```

Comment:
The name of this field is misleading: you store a JSON representation of chunks, not their positions.